modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewLocalServiceImpl.java

```diff
@@ -337,9 +345,56 @@ private void _validate(long objectViewId, long objectDefinitionId)
 		}
 	}
 
+	private void _validate(
+			ObjectView objectView, List<ObjectViewColumn> objectViewColumns)
+		throws ObjectViewColumnException {
+
+		Set<String> objectViewColumnObjectFieldNames = SetUtil.fromArray(
+			new String[0]);
+
+		for (ObjectViewColumn objectViewColumn : objectViewColumns) {
+			if (objectViewColumnObjectFieldNames.contains(
+					objectViewColumn.getObjectFieldName())) {
+
+				throw new ObjectViewColumnException(
+					"Already exist an object view column with field name: " +
+						objectViewColumn.getObjectFieldName());
+			}
+
+			objectViewColumnObjectFieldNames.add(
+				objectViewColumn.getObjectFieldName());
+		}
+
+		List<ObjectField> objectFields =
+			_objectFieldPersistence.findByObjectDefinitionId(
+				objectView.getObjectDefinitionId());
+
+		Set<String> objectFieldNames = SetUtil.fromArray(
+			new String[] {
+				"creator", "dateCreated", "dateModified", "id", "status"
+			});
+
+		for (ObjectField objectField : objectFields) {
+			objectFieldNames.add(objectField.getName());
+		}
```

Comment:
You could use like this, just to avoid another variable creation.

```
objectFieldNames.addAll(
			SetUtil.fromArray(
				"creator", "dateCreated", "dateModified", "id",	"status"));

		List<ObjectField> objectFields =
			_objectFieldPersistence.findByObjectDefinitionId(
				objectView.getObjectDefinitionId());

		for (ObjectField objectField : objectFields) {
			objectFieldNames.add(objectField.getName());
		}
```